src/qt/paymentserver.cpp

```diff
@@ -232,15 +232,18 @@ void PaymentServer::handleURIOrFile(const QString& s)
             SendCoinsRecipient recipient;
             if (GUIUtil::parseBitcoinURI(s, &recipient))
             {
-                if (!IsValidDestinationString(recipient.address.toStdString())) {
+                std::string error_msg;
+                CTxDestination dest = DecodeDestination(recipient.address.toStdString(), error_msg);
+
+                if (!IsValidDestination(dest)) {
                     if (uri.hasQueryItem("r")) {  // payment request
                         Q_EMIT message(tr("URI handling"),
                             tr("Cannot process payment request because BIP70 is not supported.\n"
                                "Due to widespread security flaws in BIP70 it's strongly recommended that any merchant instructions to switch wallets be ignored.\n"
                                "If you are receiving this error you should request the merchant provide a BIP21 compatible URI."),
                             CClientUIInterface::ICON_WARNING);
                     }
-                    Q_EMIT message(tr("URI handling"), tr("Invalid payment address %1").arg(recipient.address),
+                    Q_EMIT message(tr("URI handling"), tr(error_msg.c_str()),
```

Comment:
how would `tr(error_msg.c_str())` work for translations?

The `DecodeDestination` function will return one of many messages depending on the error of the input address. I don't think that `QTranslator` will input addresses into `DecodeDestination` to get all the error messages so that they can be translated.

Reply:
Sorry I have no clue how translation works in Bitcoin Core.

Reply:
> Sorry I have no clue how translation works in Bitcoin Core.

See https://github.com/bitcoin-core/gui/pull/280#discussion_r623790353